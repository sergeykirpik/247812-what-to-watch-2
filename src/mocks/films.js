const films = [
  {
    id: 1,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  },
  {
    id: 2,
    title: `Bohemian Rhapsody`,
    previewImage: `img/bohemian-rhapsody.jpg`,
  },
  {
    id: 3,
    title: `Macbeth`,
    previewImage: `img/macbeth.jpg`,
  },
  {
    id: 4,
    title: `Aviator`,
    previewImage: `img/aviator.jpg`,
  },
  {
    id: 5,
    title: `We need to talk about Kevin`,
    previewImage: `img/we-need-to-talk-about-kevin.jpg`,
  },
  {
    id: 6,
    title: `What We Do in the Shadows`,
    previewImage: `img/what-we-do-in-the-shadows.jpg`,
  },
  {
    id: 7,
    title: `Revenant`,
    previewImage: `img/revenant.jpg`,
  },
  {
    id: 8,
    title: `Johnny English`,
    previewImage: `img/johnny-english.jpg`,
  },
  {
    id: 9,
    title: `Shutter Island`,
    previewImage: `img/shutter-island.jpg`,
  },
  {
    id: 10,
    title: `Pulp Fiction`,
    previewImage: `img/pulp-fiction.jpg`,
  },
  {
    id: 11,
    title: `No Country for Old Men`,
    previewImage: `img/no-country-for-old-men.jpg`,
  },
  {
    id: 12,
    title: `Snatch`,
    previewImage: `img/snatch.jpg`,
  },
  {
    id: 13,
    title: `Moonrise Kingdom`,
    previewImage: `img/moonrise-kingdom.jpg`,
  },
  {
    id: 14,
    title: `Seven Years in Tibet`,
    previewImage: `img/seven-years-in-tibet.jpg`,
  },
  {
    id: 15,
    title: `Midnight Special`,
    previewImage: `img/midnight-special.jpg`,
  },
  {
    id: 16,
    title: `War of the Worlds`,
    previewImage: `img/war-of-the-worlds.jpg`,
  },
  {
    id: 17,
    title: `Dardjeeling Limited`,
    previewImage: `img/dardjeeling-limited.jpg`,
  },
  {
    id: 18,
    title: `Orlando`,
    previewImage: `img/orlando.jpg`,
  },
  {
    id: 19,
    title: `Mindhunter`,
    previewImage: `img/mindhunter.jpg`,
  },
  {
    id: 20,
    title: `Midnight Special`,
    previewImage: `img/midnight-special.jpg`,
  }
];

export default films;
