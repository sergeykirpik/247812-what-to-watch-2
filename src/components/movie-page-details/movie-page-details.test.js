import React from 'react';
import renderer from 'react-test-renderer';
import MoviePageDetails from './movie-page-details';

import films from '../../mocks/films2';

it(`renders correctly`, () => {
  const tree = renderer.create(
      <MoviePageDetails
        film={films[0]} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
