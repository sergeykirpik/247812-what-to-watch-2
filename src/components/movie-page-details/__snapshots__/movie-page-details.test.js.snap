// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "#CBAC79",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Once Upon a Time in America"
          src="https://es31-server.appspot.com/wtw/static/film/background/ones_upon_a_time_in_america.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              src="img/avatar.jpg"
              width="63"
            />
          </div>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Once Upon a Time in America
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Crime
            </span>
            <span
              className="movie-card__year"
            >
              1984
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="add-review.html"
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Once Upon a Time in America poster"
            height="327"
            src="https://es31-server.appspot.com/wtw/static/film/poster/Once_Upon_a_Time_in_America.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-card__text movie-card__row"
          >
            <div
              className="movie-card__text-col"
            >
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Director
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  Sergio Leone
                </span>
              </p>
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Starring
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  Robert De Niro,
                  <br />
                  James Woods,
                  <br />
                  Elizabeth McGovern
                </span>
              </p>
            </div>
            <div
              className="movie-card__text-col"
            >
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Run Time
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  3h 49m
                </span>
              </p>
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Genre
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  Crime
                </span>
              </p>
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Released
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  1984
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    />
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
